src/containers/editor/graph/Handle.tsx

```diff
@@ -15,10 +15,12 @@ TargetHandle.displayName = "TargetHandle";
 interface SourceHandleProps {
   id: string;
   indexInParent: number;
+  isChildrenHidden?: boolean;
```

Comment:
You should pass down `sourceId` and use `const { getNode } = useReactFlow();` inside `SourceHandle`. This is because `targetId` can be generated from `sourceId` and key (which is the `id` of `SourceHandle`). The performance will be much better than using `isChildrenHidden`, since the status change of `hidden` will only cause `SourceHandle` to re-render, rather than the entire `ObjectNode`.

Working on virtual stuff is much more complicated than on a real one, since you must always consider performance with every smill change :)

Reply:
This way seems to not occur re-rendering. Let me find a way if I can achieve this.

Reply:
@haecheonlee Oh, I think you are right. I'll accept your commits after I test it tomorrow.